hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java

```
@@ -0,0 +1,155 @@
+        return TransactionType.ETHEREUMTRANSACTION;
+    }
+
+    private EthereumTransaction insertEthereumTransaction(RecordItem recordItem, EntityId senderId) {
+        if (!entityProperties.getPersist().isEthereumTransactions()) {
+            return null;
+        }
+
+        var body = recordItem.getTransactionBody().getEthereumTransaction();
+        var ethereumDataBytes = DomainUtils.toBytes(body.getEthereumData());
+        var ethereumTransaction = ethereumTransactionParser.parse(ethereumDataBytes);
+        if (ethereumTransaction == null) {
+            return null;
+        }
+
+        // update ethereumTransaction with body values
+        var file = body.getCallData() == FileID.getDefaultInstance() ? null : EntityId.of(body.getCallData());
+        ethereumTransaction.setCallDataId(file);
+        ethereumTransaction.setMaxGasAllowance(body.getMaxGasAllowance());
+
+        // update ethereumTransaction with record values
+        var transactionRecord = recordItem.getRecord();
+        ethereumTransaction.setConsensusTimestamp(recordItem.getConsensusTimestamp());
+        ethereumTransaction.setData(ethereumDataBytes);
+        ethereumTransaction.setHash(DomainUtils.toBytes(transactionRecord.getEthereumHash()));
+        ethereumTransaction.setPayerAccountId(recordItem.getPayerAccountId());
+
+        ethereumTransaction.setFromAddress(DomainUtils.toEvmAddress(senderId));
+
+        entityListener.onEthereumTransaction(ethereumTransaction);
+        return ethereumTransaction;
+    }
+
+    private void insertContract(EntityId contractId, EntityId senderId, EthereumTransaction ethereumTransaction,
+                                long consensusTimestamp) {
+        if (!entityProperties.getPersist().isContracts()) {
+            return;
+        }
+
+        if (ethereumTransaction == null) {
+            return;
+        }
+
+        // persist newly created Contract since no ContractCreate is exposed
+        if (contractId != null) {
```

Use the fail fast approach like above. Can just add to `if (ethereumTransaction == null || contractId == null)`

---

Updated